src/ILLink.CodeFix/RequiresAssemblyFilesCodeFixProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using ILLink.RoslynAnalyzer;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace ILLink.CodeFix
+{
+	[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (RequiresAssemblyFilesCodeFixProvider)), Shared]
+	public class RequiresAssemblyFilesCodeFixProvider : BaseAttributeCodeFixProvider
+	{
+		private const string s_title = "Add RequiresAssemblyFiles attribute to parent method";
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds
+			=> ImmutableArray.Create (RequiresAssemblyFilesAnalyzer.IL3000, RequiresAssemblyFilesAnalyzer.IL3001, RequiresAssemblyFilesAnalyzer.IL3002);
+
+		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			await BaseRegisterCodeFixesAsync (context, AttributeableParentTargets.Method | AttributeableParentTargets.Property | AttributeableParentTargets.Event, RequiresAssemblyFilesAnalyzer.FullyQualifiedRequiresAssemblyFilesAttribute, s_title);
```

Comment:
nit: should we add `AttributableParentTargets.Constructor`? It doesn't matter that much for now, since both RAF and RUC can target constructors _and_ methods, but we could need more granular analysis in the future (we can ignore this for now if it isn't trivial to differentiate between methods/ctors).

Reply:
Personally I would add the complexity only when we need it, but changing from "Method" to reflect that it encompasses everything isn't bad.

Reply:
I changed the name to MethodOrConstructor